### ISSUE TITLE
Installation failed

### ISSUE THREAD

**Issue by Cargopa** [biostars/biostar-handbook-issues#278]
Hi,

So I am new to Biostar and coding. I tried setting up my computer but the automatic installation failed with fastq-dump. That was the first attempt.

Then I started doing it step by step but it failed again with the instalaltion of mamba. My partner tried to give me a hand, since he is a software engineer, but he could not fix it either. He tried to run the xargs mamba in my M1 MacBook and in docker, but got the same error message.

root@903d10c4bd40 /
# curl -s http://data.biostarhandbook.com/install/conda.txt | xargs mamba install -q -y
Could not solve for environment specs
The following packages are incompatible
├─ bcftools does not exist (perhaps a typo or a missing channel);
├─ bedtools does not exist (perhaps a typo or a missing channel);
├─ bioawk does not exist (perhaps a typo or a missing channel);
├─ blast does not exist (perhaps a typo or a missing channel);
├─ bowtie2 does not exist (perhaps a typo or a missing channel);
├─ bwa does not exist (perhaps a typo or a missing channel);
├─ csvtk does not exist (perhaps a typo or a missing channel);
├─ datamash does not exist (perhaps a typo or a missing channel);
├─ emboss does not exist (perhaps a typo or a missing channel);
├─ fastp does not exist (perhaps a typo or a missing channel);
├─ freebayes does not exist (perhaps a typo or a missing channel);
├─ hisat2 does not exist (perhaps a typo or a missing channel);
├─ mafft does not exist (perhaps a typo or a missing channel);
├─ minimap2 does not exist (perhaps a typo or a missing channel);
├─ parallel does not exist (perhaps a typo or a missing channel);
├─ samtools >=1.14  does not exist (perhaps a typo or a missing channel);
├─ seqkit does not exist (perhaps a typo or a missing channel);
├─ seqtk does not exist (perhaps a typo or a missing channel);
├─ subread does not exist (perhaps a typo or a missing channel);
└─ ucsc-bedgraphtobigwig does not exist (perhaps a typo or a missing channel).
(bioinfo)

**Comment by ialbert** [comment]
the bioconda channels have not been activated

if only fastq dump fails that is fine no need to rerun the install, the fastq dump is installed separately

we should troubleshoot only that

**Comment by Cargopa** [comment]
How about the list of incompatible packages?

**Comment by ialbert** [comment]
not sure what you mean, the "incompatibility" is that the packages do not exist because the channels have not been activated

you need to activate the correct channels.

as it happens though, just today a new set of installations have been released that use micromamba instead of conda and mamba

I have rerun and tested the new installation on both mac and Linux this morning and both ran with no problems

**Comment by Cargopa** [comment]
I did activate bioinfo channel, but it complains package not exist
```
root@e14a9ee918fc /
# conda config --add channels bioconda
Warning: 'bioconda' already in 'channels' list, moving to the top
(bioinfo) 
root@e14a9ee918fc /
# conda config --add channels conda-forge
Warning: 'conda-forge' already in 'channels' list, moving to the top
(bioinfo) 
root@e14a9ee918fc /
# conda activate bioinfo
(bioinfo) 
root@e14a9ee918fc /
# mamba install hisat2 -q -y 
Could not solve for environment specs
The following package could not be installed
└─ hisat2 does not exist (perhaps a typo or a missing channel).
(bioinfo)
```
I also tried to install using conda, but most of them didn't exist either. 
```
oot@e14a9ee918fc /
# conda install -c bioconda hisat2 -q -y 
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - hisat2

Current channels:

  - https://conda.anaconda.org/bioconda/linux-aarch64
  - https://conda.anaconda.org/bioconda/noarch
  - https://conda.anaconda.org/conda-forge/linux-aarch64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/linux-aarch64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-aarch64
  - https://repo.anaconda.com/pkgs/r/noarch
 ```

So far I have only managed to install fastqc and bio
```
root@e14a9ee918fc /
# ~/bin/doctor.py
# Doctor! Doctor! Give me the news.
# Checking symptoms ...
# bwa           --> ERROR
# datamash      --> ERROR
# fastqc -h     ... OK
# hisat2        --> ERROR
# seqret -h     --> ERROR
# featureCounts --> ERROR
# efetch        --> ERROR
# esearch       --> ERROR
# samtools      --> ERROR
# fastq-dump    --> ERROR
# bowtie2       --> ERROR
# bcftools      --> ERROR
# seqtk         --> ERROR
# seqkit        --> ERROR
# bio           ... OK
# fastq-dump -X 1 -Z SRR1553591 --> ERROR
```

I started from the scratch multiple times, not sure what went wrong, but I am sure it's something crucial.

**Comment by ialbert** [comment]
I see that you are running the commands as root. That should not be necessary.

What operating system and CPU is this?  

Rereading the post it seems you are running docker and other technologies. None of that would be necessary.

**Comment by ialbert** [comment]
Also on a Mac if you are installing conda by hand make sure to install the Intel based conda even for Arm (M1) processors. 

Bioinformatics software is only compiled for Intel based processors and the Rosetta emulation layer can run all software seamlessly.

The automated install will take care of this but if you need step by step instructions can be seen here:

https://www.biostarhandbook.com/step-by-step.html

**Comment by lizziewright** [comment]
I have a similar though not identical problem when I run curl -s http://data.biostarhandbook.com/install/conda.txt | xargs conda install -q -y
I get xargs: conda: No such file or directory
(bioinfo)
Does xargs need to be replaced with something else?  (Hi Istvan!)

**Comment by lizziewright** [comment]
ps: this is ubuntu run through WSL on my windows PC

**Comment by ialbert** [comment]
xargs reports that conda does not exist, so conda as a command is not found, 

make sure to restart the terminal and verify that conda as a command can be run

**Comment by lizziewright** [comment]
I am wondering if the problem is I can't run this script (error message) curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xj bin/micromamba to install micromamba so I resorted to these directions https://kontext.tech/article/1064/install-miniconda-and-anaconda-on-wsl-2-or-linux  I am about to install but wondering about location, does it need to go into /bin because it is defaulting to:

 Miniconda3 will now be installed into this location:

/home/liziz/miniconda3

  - Press ENTER to confirm the location
  - Press CTRL-C to abort the installation
  - Or specify a different location below

**Comment by ialbert** [comment]
make sure that you follow the linux installation instructions, most likely your linux is missing bzip packager and that is why it cannot unpack the default micromamba,

it is very, very unfortunate that micromamba developers have used a compression algorithm that may not be installed by default, but you you really only need to run

    sudo install bzip2 

then it should install. 

The default installation script should work on Linux as long as you follow the instructions from the start

**Comment by lizziewright** [comment]
conda is definitely working after I installed in the above manner but now I have this problem ./bin/micromamba shell init -s bash -p ~/micromamba -q
-bash: ./bin/micromamba: No such file or directory

**Comment by lizziewright** [comment]
how things look currently with directories

(base) liziz@lizzie:~$ ls
Miniconda3-py39_4.12.0-Linux-x86_64.sh  Miniconda3-py39_4.12.0-Linux-x86_64.sh.1  Miniconda3.sh  bin  miniconda3

**Comment by lizziewright** [comment]
bzip2 worked and installed via default installation script
# You are doing well, Majesty!
are you running course live at Penn?  I am trying to develop bioinformatics course for babies at CCNY (I'm lecturer in chem dept)

**Comment by ialbert** [comment]
yes the course in now running live, we moved to micromamba now as it is a simpler version of conda/mamba,

the words conda/mamba/micromamba are used interchangeably and technically each would work, but the book now uses micromabma as an alias for each
